adapters/tlsio_appleios.c

```diff
@@ -0,0 +1,508 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
```

Comment:
Generally looks good I believe. Thanks for addressing all the first wave comments! The only item that keeps me waiting is the dowork being called from open/close. If you could fix that I'm all good!

Cheers,
/Dan